hudi-common/src/main/java/org/apache/hudi/common/fs/HoodieWrapperFileSystem.java

```diff
@@ -1039,21 +1039,27 @@ public void createImmutableFileInPath(Path fullPath, Option<byte[]> content)
         fsout.write(content.get());
       }
     } catch (IOException e) {
-      String errorMsg = "Failed to create file" + (tmpPath != null ? tmpPath : fullPath);
+      String errorMsg = "Failed to create file " + (tmpPath != null ? tmpPath : fullPath);
       throw new HoodieIOException(errorMsg, e);
     } finally {
       try {
         if (null != fsout) {
           fsout.close();
         }
       } catch (IOException e) {
-        String errorMsg = "Failed to close file" + (needTempFile ? tmpPath : fullPath);
+        String errorMsg = "Failed to close file " + (needTempFile ? tmpPath : fullPath);
         throw new HoodieIOException(errorMsg, e);
       }
 
       try {
         if (null != tmpPath) {
-          fileSystem.rename(tmpPath, fullPath);
+          boolean renameSuccess = fileSystem.rename(tmpPath, fullPath);
+          if (!renameSuccess) {
+            fileSystem.delete(tmpPath, false);
+            LOG.error("Fail to rename " + tmpPath + " to " + fullPath);
+            throw new HoodieIOException("Failed to rename " + tmpPath + " to the target " + fullPath
+                + ", target file exists: " + fileSystem.exists(fullPath));
+          }
```

Comment:
The exception `HoodieIOException` is catched up by the outer code, can we avoid that?

Reply:
how about move `renameSuccess ` and `throw` out of  the try block

Reply:
@danny0405 @hbgstc123 hello, why not throw exception? in https://github.com/apache/hudi/pull/9212 case, if we not throw error, the multi job in the same action can create the same request file too, because rename will not throw exception

Reply:
And in use marker to solve conflict file, one job will delete other job data file